docs/src/modules/ROOT/pages/quickstart/vrp-quarkus/vehicle-routing-model.adoc

```diff
@@ -0,0 +1,790 @@
+        return visits;
+    }
+
+    public void setVisits(List<Visit> visits) {
+        this.visits = visits;
+    }
+
+    public int getTotalDemand() {
+        int totalDemand = 0;
+        for (Visit visit : visits) {
+            totalDemand += visit.getDemand();
+        }
+        return totalDemand;
+    }
+
+    public long getTotalDrivingTimeSeconds() {
+        if (visits.isEmpty()) {
+            return 0;
+        }
+
+        long totalDrivingTime = 0;
+        Location previousLocation = homeLocation;
+
+        for (Visit visit : visits) {
+            totalDrivingTime += previousLocation.getDrivingTimeTo(visit.getLocation());
+            previousLocation = visit.getLocation();
+        }
+        totalDrivingTime += previousLocation.getDrivingTimeTo(homeLocation);
+
+        return totalDrivingTime;
+    }
+
+    @Override
+    public String toString() {
+        return id;
+    }
+}
+----
+--
+
+Kotlin::
++
+--
+Create the `src/main/kotlin/org/acme/vehiclerouting/domain/Vehicle.kt` class:
+
+[source,kotlin]
+----
+package org.acme.vehiclerouting.domain
+
+import java.time.LocalDateTime
+import java.util.ArrayList
+
+import ai.timefold.solver.core.api.domain.entity.PlanningEntity
+import ai.timefold.solver.core.api.domain.lookup.PlanningId
+import ai.timefold.solver.core.api.domain.variable.PlanningListVariable
+
+@PlanningEntity
+class Vehicle {
+    @PlanningId
+    lateinit var id: String
+    var capacity: Int = 0
+    lateinit var homeLocation: Location
+    lateinit var departureTime: LocalDateTime
+
+    @PlanningListVariable
+    var visits: List<Visit>? = null
+
+    constructor()
+
+    constructor(id: String, capacity: Int, homeLocation: Location, departureTime: LocalDateTime) {
+        this.id = id
+        this.capacity = capacity
+        this.homeLocation = homeLocation
+        this.departureTime = departureTime
+        this.visits = ArrayList()
+    }
+
+    val totalDemand: Long
+        get() {
+            var totalDemand = 0L
+            for (visit in visits!!) {
+                totalDemand += visit.demand
+            }
+            return totalDemand
+        }
+
+    val totalDrivingTimeSeconds: Long
+        get() {
+            if (visits!!.isEmpty()) {
+                return 0
+            }
+
+            var totalDrivingTime: Long = 0
+            var previousLocation = homeLocation
+
+            for (visit in visits!!) {
+                totalDrivingTime += previousLocation.getDrivingTimeTo(visit.location!!)
+                previousLocation = visit.location!!
+            }
+            totalDrivingTime += previousLocation.getDrivingTimeTo(homeLocation)
+
+            return totalDrivingTime
+        }
+
+    override fun toString(): String {
+        return id
+    }
+}
+----
+--
+====
+
+The `Vehicle` class has an `@PlanningEntity` annotation, so Timefold Solver knows that this class changes during solving
+because it contains one or more planning variables.
```

Comment:
I used this paragraph as a reference to write the one above.

"The Lesson class has an @PlanningEntity annotation, so Timefold Solver knows that this class changes during solving because it contains one or more planning variables."

Reply:
Let's keep it for now, we can always remove later.